### PR TITLE
Feature/multi sig

### PR DESCRIPTION
**Pull request by Jedi2002**
The PR contains the basic implementation for creating an airdrop and related queries . 

An additional feature for this module has been implemented in 
https://github.com/Sifchain/sifnode/pull/1082

### CODE REVIEW

**Review by intl-man — APPROVED** [review]
@Jedi2002 this LGTM. I've gone through it a few times and can't see anything that raises red flags. Would like to get this merged and deployed into DevNet sooner rather than later.

File: x/dispensation/types/records.go [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+		return "Pending"
+	case Completed:
+		return "Completed"
+	default:
+		return "All"
+	}
+}
+
+// DistributionRecord is created for every recipient for a distribution
+// TODO : Remove ClaimStatus from this struct and use it as prefix.
```
Comment (Jedi2002) [inline x/dispensation/types/records.go:28]: Or keep it in both places

**Review by intl-man — APPROVED** [review]

File: app/poolChange.go [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+				break
+			}
+			poolList = append(poolList, pool)
+		}
+		// If we have error dont set state
+		if hasError {
+			ctx.Logger().Error("Failed to execute upgrade plan for pool re-balance")
+		}
+		// If we have no errors , Set state .
+		if !hasError {
```
Comment (timlind) [inline app/poolChange.go:61]: Using sub functions that returns errors might make this a bit cleaner / easier to follow.

**Review by timlind — APPROVED** [review]

**Review by iverc — APPROVED** [review]
🚀

File: app/poolChange.go (hunk 2 above)
Comment (Jedi2002) [reply app/poolChange.go:61]: Yeah that would have been the better thing to do here . 
But this is an old piece of code ,and the upgrade for this has already been executed . Just keeping it here for historical records .

**Pull request merged**